src/plugins/explore/public/components/top_nav/top_nav.tsx

```diff
@@ -171,27 +182,59 @@ export const TopNav = ({ setHeaderActionMenu = () => {}, savedExplore }: TopNavP
     [queryString, data.dataViews, dispatch, services, clearEditors]
   );
 
+  // Custom onChange handler to track date range changes in Redux (mirrors SearchBar behavior)
+  const handleQueryChange = useCallback(
+    (queryAndDateRange: { dateRange: any; query?: Query }) => {
+      if (queryAndDateRange.dateRange) {
+        dispatch(setDateRange(queryAndDateRange.dateRange));
+      }
+    },
+    [dispatch]
+  );
+
+  const handleQuerySubmit = useCallback(() => {
+    dispatch(onEditorRunActionCreator(services, editorText));
+    dispatch(setQueryExecutionButtonStatus('REFRESH'));
```

Comment:
will `onEditorRunActionCreator` always set the execution button status to REFRESH? If so, can we add `dispatch(setQueryExecutionButtonStatus('REFRESH'))` to the end of `onEditorRunActionCreator`?

Reply:
This is to reset the button status. Could be in onEditorRunActionCreator